includes/class-sensei-course.php

```diff
@@ -100,8 +100,9 @@ public function __construct () {
         add_action('save_post', array( 'Sensei_Course', 'flush_rewrite_rules' ) );
 
 		// Allow course archive to be setup as the home page
-		add_action( 'pre_get_posts', array( $this, 'allow_course_archive_on_front_page' ) );
-
+		if ( (int) get_option( 'page_on_front' ) > 0 ) {
```

Comment:
Giving that `page_on_front` _should_ always be a post id in order to be valid, this makes sense to me.